database/migrations/2020_09_18_1800_add_missing_primary_keys.php

```diff
@@ -0,0 +1,92 @@
+<?php
+
+/*
+    This migration adds all missing primary keys for tables.
+
+    Percona Xtradb refuses to INSERT into a table
+    without a primary key.
+ */
+
+use Illuminate\Database\Migrations\Migration;
+use Illuminate\Database\Schema\Blueprint;
+
+class AddMissingPrimaryKeys extends Migration
+{
+
+    /**
+     * Run the migrations.
+     *
+     * @return void
+     */
+    public function up()
+    {
+        if (!Schema::hasColumn('bill_perms', 'id')) {
+            Schema::table('bill_perms', function (Blueprint $table) {
+                $table->bigIncrements('id')->first();
+            });
+        }
+
+        if (!Schema::hasColumn('bill_ports', 'id')) {
+            Schema::table('bill_ports', function (Blueprint $table) {
+                $table->bigIncrements('id')->first();
+            });
+        }
+
+        if (!Schema::hasColumn('devices_perms', 'id')) {
+            Schema::table('devices_perms', function (Blueprint $table) {
+                $table->bigIncrements('id')->first();
+            });
+        }
+
+        if (!Schema::hasColumn('entPhysical_state', 'id')) {
+            Schema::table('entPhysical_state', function (Blueprint $table) {
+                $table->bigIncrements('id')->first();
+            });
+        }
+
+        if (!Schema::hasColumn('ipv4_mac', 'id')) {
+            Schema::table('ipv4_mac', function (Blueprint $table) {
+                $table->bigIncrements('id')->first();
+            });
+        }
+
+        if (!Schema::hasColumn('juniAtmVp', 'id')) {
+            Schema::table('juniAtmVp', function (Blueprint $table) {
+                $table->bigIncrements('id')->first();
+            });
+        }
+
+        if (!Schema::hasColumn('loadbalancer_vservers', 'id')) {
+            Schema::table('loadbalancer_vservers', function (Blueprint $table) {
+                $table->bigIncrements('id')->first();
+            });
+        }
+
+        if (!Schema::hasColumn('ports_perms', 'id')) {
+            Schema::table('ports_perms', function (Blueprint $table) {
+                $table->bigIncrements('id')->first();
+            });
+        }
+
+        if (!Schema::hasColumn('processes', 'id')) {
+            Schema::table('processes', function (Blueprint $table) {
+                $table->bigIncrements('id')->first();
+            });
+        }
+
+        if (!Schema::hasColumn('transport_group_transport', 'id')) {
+            Schema::table('transport_group_transport', function (Blueprint $table) {
+                $table->bigIncrements('id')->first();
+            });
+        }
+    }
+
+    /**
+     * Reverse the migrations.
+     *
+     * @return void
+     */
+    public function down()
+    {
```

Comment:
Hi
It seems that you did not implement the "reverse" part of the migration. The idea is to be exactly back to the initial status after reverting a migration php file. This is one of the reason we can do "live" updates in a safer way ...

Reply:
@PipoCanaja That is true. I found that my earlier pull request which originally contained a reverse migration step was changed by a developer, to not contain that anymore, and I've just followed that way. My earlier PR was #11331, and the file I've created then now dont have a reverse step too:

https://github.com/librenms/librenms/blob/master/database/migrations/2020_03_24_0844_add_primary_key_to_device_graphs.php

How to proceed now?

Reply:
Right. @murrant : What do you suggest ? You removed the down() code for compatibility with sqlite ?

Reply:
@PipoCanaja @murrant ping

Reply:
@PipoCanaja @murrant please comment my work.

Reply:
@rkojedzinszky Let's have @murrant opinion on this. 

- Aside of this, you could sign the CLA (this is a prerequisite).
- You should also rebase your patch, as a lot of changes took place in the last few days. This will most probably solve the error in TravisCI.

Reply:
we don't really use down and sqlite cannot remove indexes without recreating the table.

Reply:
> @rkojedzinszky why do you have the hasColumn checks? If you are concerned about the migration failing mid migration, you should create one migration for each table affected.

@murrant I've updated the patch. Basically I've just used my earlier modified migration step as a template here. Here the migration should run without errors, so no checks are needed. Do you like this way, or should I split into separate migration steps?

Reply:
It is 11 tables and we have been burned by a bad migration before.  Probably best to split it up.  Thanks.